enumerable.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...to_a.length).each do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
```

Comment:
- [ ] identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)). returns an enumerator if no block is given. You can achieve this by adding `return to_enum`